pkg/controller/daemon/update.go

```diff
@@ -38,7 +38,7 @@ func (dsc *DaemonSetsController) rollingUpdate(ds *extensions.DaemonSet) error {
 		return fmt.Errorf("couldn't get node to daemon pod mapping for daemon set %q: %v", ds.Name, err)
 	}
 
-	_, oldPods, err := dsc.getAllDaemonSetPods(ds, nodeToDaemonPods)
```

Comment:
If getAllDaemonSetPods is always returning `nil`  just change the function itself to not return an error.